### DISCUSSION TITLE
What's the plan for P2P module?

### DISCUSSION THREAD

**Discussion by worldisreal** [tendermint/tendermint#9711] [Q&A] [answered]
I see there is an implementation of ADR-62 in master branch. However, in https://github.com/tendermint/tendermint/issues/5670 I also noticed the refactor was interrupted. 

My question is -  will ADR-62 in the master branch be released or not, and what's the timeline? libp2p is also on the table, any timeline of it?

Thanks.

**Comment by worldisreal** [comment]
@cmwaters can you look this when you get time? thanks

**Comment by thanethomson** [comment, marked as answer]
It's unclear as to whether we'll actually be adopting libp2p. At least in the short-term (3-6 months), there's no work planned to do so any more, especially given the issues encountered with the P2P layer in v0.35 (see #9155).

We're in the process of specifying the behaviour that we want from the P2P layer in #9573. We may eventually adopt parts of #5670, depending on our spec work, but it's uncertain at this point.

**Reply by worldisreal** [reply]
Thanks a lot for your reply. Cool.
